libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/VoiceInstructionLoader.java

```diff
@@ -0,0 +1,85 @@
+package com.mapbox.services.android.navigation.v5.navigation;
+
+import com.mapbox.api.directions.v5.models.LegStep;
+import com.mapbox.api.directions.v5.models.VoiceInstructions;
+import com.mapbox.api.speech.v1.MapboxSpeech;
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import okhttp3.ResponseBody;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class VoiceInstructionLoader {
+  private static VoiceInstructionLoader instance;
+  private MapboxSpeech.Builder mapboxSpeech;
+
+  private VoiceInstructionLoader() {
+
+  }
+
+  public static synchronized VoiceInstructionLoader getInstance() {
```

Comment:
Could you remind me why we need the singleton pattern here?

Reply:
@Guardiola31337 so that we can call it from the voice instruction milestone to cache upcoming instructions, but call it from MapboxSpeechPlayer when it's time to actually voice the instruction